ApiBundle/Transformer/TemplateTransformer.php

```diff
@@ -42,14 +43,18 @@ public function transform($template)
             array('templateId' => $template->getTemplateId())
         ));
 
-        $facade->addLink('_self_delete', $this->generateRoute('open_orchestra_api_template_delete',
-            array('templateId' => $template->getTemplateId())
-        ));
+        if ($this->authorizationChecker->isGranted(TreeTemplatePanelStrategy::ROLE_ACCESS_DELETE_TEMPLATE)) {
+            $facade->addLink('_self_delete', $this->generateRoute('open_orchestra_api_template_delete',
+                array('templateId' => $template->getTemplateId())
+            ));
+        }
 
-        $facade->addLink('_self_update_areas', $this->generateRoute('open_orchestra_api_areas_update_in_template',
-            array(
-                'templateId' => $template->getTemplateId())
-        ));
+        if ($this->authorizationChecker->isGranted(TreeTemplatePanelStrategy::ROLE_ACCESS_UPDATE_TEMPLATE)) {
```

Comment:
As in the `AreaTransformer` you should pass the `$template` to the `isGranted` method